### PR TITLE
add `hash` option

### PR DESCRIPTION
**Pull request by stephenmathieson**
This patch adds an optional `hash` option to the API. You may use it to
create custom hashing for requests.

My use case is to ignore specific (cache-busting) query parameters, but
you could implement nearly anything here.

See #29.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -60,18 +60,20 @@ module.exports = function (host, opts) {
...
+  function tapename(req, body) {
+    var hash = opts.hash || messageHash.sync;
 
-function tapename(req, body) {
-  return hash.sync(req, Buffer.concat(body)) + '.js';
-}
+    return hash(req, Buffer.concat(body)) + '.js';
+  }
+
+};
```
Comment (jeremyruppel) [inline index.js]: The only question I have about this is this spacing here. Why the change?

Comment (jeremyruppel) [inline index.js]: Obviously whitespace isn't a blocker for shipping this though 😄

**Comment by jeremyruppel** [comment]
This looks awesome! Thanks for the patch!

**Comment by jeremyruppel** [comment]
This is now available in yakbak@2.6.0

File: index.js (hunk 1 above)
Comment (stephenmathieson) [inline index.js]: my only response: ¯\\_(ツ)_/¯

**Comment by stephenmathieson** [comment]
Awesome, thanks a ton!

File: index.js (hunk 1 above)
Comment (jeremyruppel) [inline index.js]: 🤣  fair enough!
